Add Luhn example (#44)
zkpstandard/wg-plonkish: src/example.md

# Example: Luhn algorithm

This page provides a detailed walkthrough of building a Plonk constraint system using the Luhn algorithm as an example. We will demonstrate how to represent a simple computation as a circuit and how to apply different types of Plonk constraints: copy constraints, lookup constraints and custom constraints.

## Luhn algorithm

The Luhn algorithm is a check digit formula used to validate identification numbers such as credit card numbers or national identification numbers. It works as follows:

> 1. Split the number into a payload (all digits except the last one) and a check digit (tthe last digit).

Reviewer: ```suggestion
1. Split the number into a payload (all digits except the last one) and a given check digit (the last digit).
```

> 2. Compute the check digit from the payload.

Reviewer: ```suggestion
2. Compute what the check digit should be from the payload:
```

   a. Starting from the right (excluding the check digit), double every second digit. If doubling results in a number greater than 9, subtract 9.

>    b. Sum all the digits (original and transformed).

Reviewer: ```suggestion
   b. Let $s$ be the sum of all the digits (transformed or not) resulting from step a.
```
